var/spack/repos/builtin/packages/mariadb/package.py

```diff
@@ -44,3 +49,6 @@ class Mariadb(CMakePackage):
     depends_on('msgpack-c')
 
     conflicts('%gcc@9.1.0:', when='@:5.5')
+
+    conflicts('platform=darwin',
+              msg='mariadb depends on libaio which does not support Darwin')
```

Comment:
I would replace all of these conflicts with:
```python
depends_on('libaio', when='platform=linux')
```
It would be a real shame if none of these packages could be built on macOS.

Reply:
`lvm2` seems to require `libaio` (I tried commenting out the dependency and it errors at configure looking for it). 

`fio` has a variant to activate the support so I left it as it is. Let me know if you think it's worthwhile removing the variant and build `libaio` conditionally on the platform.